source/objects/RenderTarget.cpp

```diff
@@ -247,6 +247,17 @@ void RenderTarget::GenerateMIPs()
 	}
 }
 
+IRenderTarget* RenderTarget::Clone() {
+	RenderTarget* rt = new RenderTarget(GetWidth(), GetHeight(), GetFormat(), mMips);
+
+	// Copy all of the mip levels (including the raw image, 0)
+	for (uint8_t mip = 0; mip < mMips; mip++) {
+		memcpy(rt->GetRawData(mip), GetRawData(mip), GetWidth(mip) * GetHeight(mip) * mPixelSize);
```

Comment:
One of the slowest ways to copy the data between them you could’ve thought of.

Why aren’t you copying **all** of the data at once using mSize, you can access private members of other class instances in a member function.